lib/spack/spack/test/build_environment.py

```diff
@@ -541,3 +541,13 @@ def test_setting_attributes(self, default_mock_concretization):
             if current_module == spack.package_base:
                 break
             assert current_module.SOME_ATTRIBUTE == 1
+
+
+def test_monkey_patching_works_across_virtual(default_mock_concretization):
```

Comment:
Why not use monkeypatch if the purpose is to test monkeypatch?

Reply:
Because we want to test monkey-patching a package like a user would do, not with an internal mechanism used by pytest.